lib/account/account-service.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+
+/**
+ * What we initially send back to the client so they can start to initiate
+ * the complete procedure and save information.
+ *
+ * @typedef {accountService~initiateResult}
+ * @param {string} accountId
+ * @param {string} email
+ * @param {string} mfa
+ * @param {string} salt
+ */
+
+/**
+ * Handles account creation from initiating to confirmation.
+ */
+class AccountService {
+    /**
+     * Used for Dependency Injection and initial configuration
+     * of the storage plugin.
+     *
+     * @param {Object} config
+     * @param {Object} password
+     * @param {Object} storage
+     */
+    constructor(config, password, storage) {
+        this.password = password;
+        this.storage = storage.configure(config.storage);
+    }
+
+
+    /**
+     * Completes the sign up of the account with more information from the
+     * client. We need the client to create the password using their tools
+     * so we never see the plain text version of it. This helps to keep
+     * the system more secure
+     *
+     * TODO: Encrypt data
+     *
+     * @param {string} directory
+     * @param {Object} accountInfo
+     * @param {Object} accountFile
+     * @param {Object} options
+     * @return {Promise.<boolean>}
+     */
+    complete(directory, accountInfo, accountFile, options) {
```

Comment:
Per my earlier comment, I don't want the accountFile added back here, otherwise the manager might want to set properties itself and not bother calling the service.